src/annotator/anchoring/range-normalized.js

```diff
@@ -0,0 +1,174 @@
+        break;
+      }
+    }
+    return this;
+  }
+
+  /**
+   * Convert this range into an object consisting of two pairs of (xpath,
+   * character offset), which can be easily stored in a database.
+   *
+   * root -           The root Element relative to which XPaths should be calculated
+   * ignoreSelector - A selector String of elements to ignore. For example
+   *                  elements injected by the annotator.
+   *
+   * Returns an instance of SerializedRange.
+   */
+  serialize(root, ignoreSelector) {
+    const serialization = function (node, isEnd) {
+      let origParent;
+      if (ignoreSelector) {
+        origParent = $(node).parents(`:not(${ignoreSelector})`).eq(0);
+      } else {
+        origParent = $(node).parent();
+      }
+      const xpath = xpathFromNode(origParent, root)[0];
+      const textNodes = getTextNodes(origParent);
+      //console.log('3')
+      // Calculate real offset as the combined length of all the
+      // preceding textNode siblings. We include the length of the
+      // node if it's the end node.
+      const nodes = textNodes.slice(0, textNodes.index(node));
+      let offset = 0;
+      for (let n of Array.from(nodes)) {
+        offset += n.nodeValue.length;
+      }
+
+      if (isEnd) {
+        return [xpath, offset + node.nodeValue.length];
+      } else {
+        return [xpath, offset];
+      }
+    };
+
+    const start = serialization(this.start);
+    const end = serialization(this.end, true);
+
+    return new SerializedRange({
+      // XPath strings
+      start: start[0],
+      end: end[0],
+      // Character offsets (integer)
+      startOffset: start[1],
+      endOffset: end[1],
+    });
+  }
+
+  /**
+   * Public: Creates a concatenated String of the contents of all the text nodes
+   * within the range.
+   *
+   * Returns a String.
+   */
+  text() {
+    return Array.from(this.textNodes())
+      .map(node => node.nodeValue)
+      .join('');
+  }
+
+  /**
+   * Public: Fetches only the text nodes within the range.
+   *
+   * Returns an Array of TextNode instances.
+   */
+  textNodes() {
+    const textNodes = getTextNodes($(this.commonAncestor));
+    const [start, end] = Array.from([
+      textNodes.index(this.start),
+      textNodes.index(this.end),
+    ]);
+    // Return the textNodes that fall between the start and end indexes.
+    return $.makeArray(textNodes.slice(start, +end + 1 || undefined));
+  }
+
+  /**
+   * Public: Converts the Normalized range to a native browser range.
+   *
+   * See: https://developer.mozilla.org/en/DOM/range
+   *
+   * Examples
+   *
+   *   selection = window.getSelection()
+   *   selection.removeAllRanges()
+   *   selection.addRange(normedRange.toRange())
+   *
+   * Returns a Range object.
+   */
+  toRange() {
+    const range = document.createRange();
+    range.setStartBefore(this.start);
+    range.setEndAfter(this.end);
+    return range;
+  }
+}
+
+export const Normalized = {
```

Comment:
I wrapped this in a namespace because `require('./range-normalized').default` was not working with the ES6 class in `range-browser.js`